libafl/src/mutators/tuneable.rs

```diff
@@ -416,44 +416,36 @@ mod test {
         let input = BytesInput::new(vec![42]);
 
         // Basic tests over the probability distribution.
-        assert!(
-            TuneableScheduledMutator::set_mutation_probabilities(&mut state, vec![0.0]).is_err()
-        );
-        assert!(
-            TuneableScheduledMutator::set_mutation_probabilities(&mut state, vec![1.0; 3]).is_err()
-        );
-        assert!(TuneableScheduledMutator::set_mutation_probabilities(
-            &mut state,
-            vec![-1.0, 1.0, 1.0]
-        )
-        .is_err());
-        assert!(TuneableScheduledMutator::set_mutation_probabilities(&mut state, vec![]).is_err());
-
-        assert!(TuneableScheduledMutator::set_mutation_probabilities(
-            &mut state,
-            vec![0.0, 0.0, 1.0]
-        )
-        .is_ok());
+        assert!(tuneable
```

Comment:
Not sure If I already commented this, but I fear this will make it really hard to use tuneable, since you have to put this into the stage tuple (=give up ownership)...

Reply:
i think this should not be associated function then. we cannot make this guy NOT to take self because then type parameter MT can't be inferred

Reply:
Just checked my code: I can probably get around it by querying the stage by type/ref(?), I do have the stages_tuple available, so it might be fine